operator/pkg/controllers/master/master.go

```diff
@@ -69,28 +77,76 @@ func (c *Controller) createSecrets(ctx context.Context, controlPlane *v1alpha1.C
 	return nil
 }
 
-func certListFor(controlPlane *v1alpha1.ControlPlane) []*secrets.Request {
+func (c *Controller) getControlPlaneHostname(ctx context.Context, controlPlane *v1alpha1.ControlPlane) (string, error) {
```

Comment:
Seems odd to me that the "get" method would "create" internally. 

I might model this like:
```
service, err := c.client.ensure(ctx, c.ServiceFor(controlPlane))
hostname := service.Status.Loadbalancer.Ingress
if  == nil {
  return result.Wait()
}

c.SecretsFor(hostname)
...
```